pkg/virt-handler/vm.go

```diff
@@ -1284,6 +1284,18 @@ func (d *VirtualMachineController) calculateLiveMigrationCondition(vmi *v1.Virtu
 	if err := d.isHostModelMigratable(vmi); err != nil {
 		return newNonMigratableCondition(err.Error(), v1.VirtualMachineInstanceReasonCPUModeNotMigratable), isBlockMigration
 	}
+	if vmi.Spec.Domain.CPU != nil {
+		if vmi.Status.TopologyHints == nil || vmi.Status.TopologyHints.TSCFrequency == nil {
+			for _, feature := range vmi.Spec.Domain.CPU.Features {
+				if feature.Name == "invtsc" {
+					switch feature.Policy {
+					case "require", "force", "optional":
+						return newNonMigratableCondition("InvTSC CPU feature disables migration support", v1.VirtualMachineInstanceReasonInvTSCFeatureNotMigratable), isBlockMigration
+					}
+				}
+			}
+		}
+	}
```

Comment:
is this possible now that the VMI won't get scheduled unless the hints exist?

Reply:
I am not sure how practically relevant it is, but if someone has a VM running nowadays it would set the right condition. However it would fail migrations nowadays too already. Will remove it.